jetty-http2/http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Stream.java

```diff
@@ -235,12 +235,15 @@ public boolean isRemotelyClosed()
     @Override
     public boolean failAllData(Throwable x)
     {
-        List<DataEntry> copy;
+        List<DataEntry> copy = List.of();
         try (AutoLock l = lock.lock())
         {
             dataDemand = 0;
-            copy = new ArrayList<>(dataQueue);
-            dataQueue.clear();
+            if (dataQueue != null)
+            {
+                copy = new ArrayList<>(dataQueue);
+                dataQueue.clear();
```

Comment:
why not take the queue? Is the stream likely to be reused after a fail? will it need a queue?
```suggestion
                copy = dataQueue;
                dataQueue = null;
```